misc/language-tour-guide/src/utils/storage/index.ts

```diff
@@ -0,0 +1,9 @@
+export function hashPathname(pathname: string): string {
+  let hash = 0;
+  for (let i = 0; i < pathname.length; i++) {
+    const char = pathname.charCodeAt(i);
+    hash = (hash << 5) - hash + char;
+    hash = hash & hash;
+  }
+  return Math.abs(hash).toString(36);
+}
```

Comment:
Good enough for now but I'd consider using some standardized non-crypto hash function like xxhash or murmur.